File: 1108.defanging-an-ip-address.java
/*
 * @lc app=leetcode id=1108 lang=java
 *
 * [1108] Defanging an IP Address
 */

// @lc code=start
class Solution {
    public String defangIPaddr(String address) {
        StringBuffer sb = new StringBuffer();
        for (char ch: address.toCharArray()) {
            if (ch >= '0' && ch <= '9') {
                sb.append(ch);
            }
            if (ch == '.') {
                sb.append("[.]");
            }
        }
        return sb.toString();
    }
}
// @lc code=end

File: 1108.defanging-an-ip-address.cpp
/*
 * @lc app=leetcode id=1108 lang=cpp
 *
 * [1108] Defanging an IP Address
 */

// @lc code=start
#include <string>

using namespace std;

class Solution {
public:
    string defangIPaddr(string address) {
        string ret(address.size() + 6, '\0');
        auto iter = ret.begin();
        for (const auto &ch : address) {
            if (ch == '.') {
                *(iter++) = '['; *(iter++) = '.'; *(iter++) = ']';
            } else {
                *(iter++) = ch;
            }
        }
        return ret;
    }
};
// @lc code=end
